test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -727,6 +782,118 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
                 .ThenBy(item => item.Key.Name, StringComparer.Ordinal));
         }
 
+        private DataAnnotationsMetadataProvider CreateLocalizingProvider()
+        {
+            var stringLocalizer = new Mock<IStringLocalizer>(MockBehavior.Strict);
+            stringLocalizer
+                .Setup(loc => loc[It.IsAny<string>()])
+                .Returns<string>((k =>
+                {
+                    if (k.Contains("Loc_Two"))
+                    {
+                        return new LocalizedString(k, $"{k} {CultureInfo.CurrentCulture}");
```

Comment:
You're testing internal details of the `IStringLocalizer` again. The caller gets a string and deals with it. None of those callers check `ResourceNotFound`